### DISCUSSION TITLE
Properties from DataRequest to DataFlowRequest is missing

### DISCUSSION THREAD

**Discussion by hlange** [eclipse-edc/Connector#1918] [Q&A] [answered]
Hello everybody,
I was working on the problem to send additional information from consumer to provider with the data-plane-http extension. During my research, I was looking at the DataFlowRequest object that has the attribute properties, however it was empty even if I put something in my request. Then I began my search to find the place where this object was created, which I then found in the DataPlaneTransferFlowController. 

It looks like that the call `createRequest `is missing  `.properties(dataRequest.getProperties())` which would allow accessing the provided properties in the request. 

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/62478e319b33018a403541baa9eb2ef2f70cbd28/extensions/control-plane/data-plane-transfer/data-plane-transfer-client/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/flow/DataPlaneTransferFlowController.java#L66-L76

With the above mentioned change the DataFlowRequest has the properties that are send by the following request.

```
{
  "protocol": "ids-multipart",
  "assetId": "asset-1",
  "contractId": "2:8b1c092e-fbea-4f94-8c09-1c23d307dc02",
  "dataDestination": {
    "properties": {
        "baseUrl":"http://127.0.0.1:8001",
        "type": "HttpData",
        "header:x-ms-blob-type": "BlockBlob"
    }
  },
  "properties" : {
    "type" : "object"
  },
  "transferType": {
    "contentType": "application/octet-stream",
    "isFinite": true
  },
  "managedResources": false,
  "connectorAddress": "http://192.168.0.6:8282/api/v1/ids/data",
  "connectorId": "consumer"
}
```

My question is, if this was intended or missing this call was a bug.

Best regards,
Henrik

**Comment by ronjaquensel** [comment, marked as answer]
The properties should be mapped. I created a respective [issue](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1972) from this discussion.
